client/my-sites/stats/stats-subscribers-page/subsriber-highlights.jsx

```diff
@@ -0,0 +1,97 @@
+import { HighlightCard, Popover } from '@automattic/components';
+import { Icon, info } from '@wordpress/icons';
+import { translate } from 'i18n-calypso';
+import { useRef, useState } from 'react';
+import '../wordads/highlights-section.scss';
+
+function getSubscriberHighlights() {
+	const highlights = [
+		{
+			id: 1,
+			heading: 'Total email subscribers',
+			count: 1410,
+			previousCount: 1080,
+		},
+		{
+			id: 2,
+			heading: 'Free email subscribers',
+			count: 1340,
+			previousCount: 1034,
+		},
+		{
+			id: 3,
+			heading: 'Paid email subscribers',
+			count: 70,
+			previousCount: 46,
+		},
+		{
+			id: 4,
+			heading: 'WordPress.com subscribers',
+			count: 322,
+			previousCount: 326,
+		},
+	];
+	return highlights;
+}
+
+function SubscriberHighlightsHeader() {
+	const [ isTooltipVisible, setTooltipVisible ] = useState( false );
+	const infoReferenceElement = useRef( null );
+	const localizedTitle = translate( 'All time stats', {
+		comment: 'Heading for Subscribers page highlights section',
+	} );
+	const showNotices = true;
+	const tooltipContent = <p>Tooltip content goes here.</p>;
+	return (
+		<h1 className="highlight-cards-heading">
+			{ localizedTitle }{ ' ' }
+			{ showNotices && (
+				<>
+					<span
+						className="info-wrapper"
+						ref={ infoReferenceElement }
+						onMouseEnter={ () => setTooltipVisible( true ) }
+						onMouseLeave={ () => setTooltipVisible( false ) }
+					>
+						<Icon className="info-icon" icon={ info } />
+					</span>
+					<Popover
+						className="tooltip tooltip--darker tooltip-wordads highlight-card-tooltip"
+						isVisible={ isTooltipVisible }
+						position="bottom right"
+						context={ infoReferenceElement.current }
+					>
+						<div className="highlight-card-tooltip-content">{ tooltipContent }</div>
+					</Popover>
+				</>
+			) }
+		</h1>
+	);
+}
+
+function SubscriberHighlightsListing() {
```

Comment:
All of the functions here are separate components. Let's use arrow functions while defining them.

Reply:
I can definitely do this. Is there a technical reason for this suggestion or is this just a preference?

In my reading, I'm not finding anything specific to React that would argue for this as a best practice. My understanding of the potentially relevant issues here are as follows:

1. Arrow functions automatically bind `this` to the calling scope. My understanding is this is only relevant in Class-based components.
2. Arrow functions are allocated at run-time each time they are called. I'm assuming this means they are copied to the stack and addressed by their memory address. This means they could actually introduce performance concerns if called repeatedly in a tight loop. Probably not relevant to our usage.
3. Related to the above, arrow functions are not "comparison safe". So passing an arrow function as a prop would always result in a new render even though the callback itself has not changed.

I'm guessing our build tools might also impact this if they are rewriting our arrow functions to some baseline version for compatibility.

Anyway, mostly just asking to make sure I'm clear on this.

cc: @grzegorz-cp

Reply:
In this circumstance, it was just a suggestion and preference.

Reply:
I normally would use normal functions for components. I'd like to know more about this too 😄 

But @a8ck3n I have some questions regarding the 'potentially relevant issues' regarding arrow functions,

> Arrow functions automatically bind this to the calling scope. My understanding is this is only relevant in Class-based components.

I'm not too sure about this statement. As I see it, the best thing about arrow functions is that they do NOT automatically bind to the calling scope.

> Arrow functions are allocated at run-time each time they are called.

Can I ask for reference regarding the statement? From my understanding, arrow functions are exactly the same as functions, they don't get allocated any way at run-time, which is only related to where you define it.

For example, if you define the function outside components, where they could just stay what they are and not re-allocated at all on re-rendering.

So I think it's not related to arrow functions themselves, but how you use them.

> Related to the above, arrow functions are not "comparison safe".

The same thing applies here too.